### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by jrha**
* Correct line numbering
* Improve debug logging output
* Remove markdown links and URL strings from text
  * Strip whitepace for tidiness.
* Ensure dictionary functionality is also tested.

### CODE REVIEW

File: src/markspelling.py [hunk 1]
```diff
@@ -71,6 +78,7 @@ def checklinelist(self, linelist, filename):
         errorcount = 0
         incodeblock = False
         for linenumber, line in enumerate(linelist):
+            linenumber += 1
```
Comment (tofu-rocketry) [inline src/markspelling.py]: Can you not use the `start` argument to `enumerate`?

File: src/markspelling.py [hunk 2]
```diff
@@ -47,17 +49,22 @@ def checkline(self, line, linenumber, filename, incodeblock=False):
         incodeblock = self.checkcodeblock(line, incodeblock)
         errorwords = list()
         if not wasincodeblock and not incodeblock:
-            self.logger.debug('Checking line "%s"', line.rstrip())
+            self.logger.debug('%03d   RAW %s', linenumber, line.rstrip())
```
Comment (tofu-rocketry) [inline src/markspelling.py]: Why a triple space?

File: src/markspelling.py [hunk 3]
```diff
@@ -47,17 +49,22 @@ def checkline(self, line, linenumber, filename, incodeblock=False):
...
             line = self.regexhtmldirty.sub('', line)  # strip html tags
             line = self.regexhtmlclean.sub('', line)  # strip inline code
+            line = self.regexlink.sub(r'\1', line)  # strip links
+            line = self.regexurl.sub('', line)  # strip URLs
+            line = line.strip()
+            self.logger.debug('%03d CLEAN %s', linenumber, line.rstrip())
             self.spellcheck.set_text(line)
             for err in self.spellcheck:
-                self.logger.debug("'%s' not found in main dictionary", err.word)
+                self.logger.debug("%03d  WARN '%s' not found in main dictionary", linenumber, err.word)
```
Comment (tofu-rocketry) [inline src/markspelling.py]: Double space on this one.

File: src/markspelling.py [hunk 4]
```diff
@@ -47,17 +49,22 @@ def checkline(self, line, linenumber, filename, incodeblock=False):
...
                 if not self.pwl or not self.pwl.check(err.word):
+                    self.logger.debug("%03d ERROR '%s' not found in custom dictionary", linenumber, err.word)
                     errorcount += 1
                     errorwords.append(err.word)
         else:
-            self.logger.debug('Skipping line "%s"', line.rstrip())
+            self.logger.debug('%03d  CODE %s', linenumber, line.rstrip())
```
Comment (tofu-rocketry) [inline src/markspelling.py]: Double space.

File: .travis.yml [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
-- 3.4
+- 3.5
 
 addons:
   apt:
     packages:
     - python3-enchant
 
+install:
+  - pip install -r requirements.txt
```
Comment (tofu-rocketry) [inline .travis.yml]: This is the default install command. You need it explicit?

File: src/markspelling.py (hunk 2 above)
Comment (jrha) [reply src/markspelling.py]: Formatting, debugging is easier when the output is lined up.

Comment (tofu-rocketry) [reply src/markspelling.py]: Ah, right.

File: src/markspelling.py [hunk 6]
```diff
@@ -47,17 +49,22 @@ def checkline(self, line, linenumber, filename, incodeblock=False):
         incodeblock = self.checkcodeblock(line, incodeblock)
         errorwords = list()
         if not wasincodeblock and not incodeblock:
-            self.logger.debug('Checking line "%s"', line.rstrip())
+            self.logger.debug('%03d %5s \033[1;30m%s\033[0m', linenumber, 'RAW', line.rstrip())
```
Comment (tofu-rocketry) [inline src/markspelling.py]: How does this work? Was more readable before.

Comment (jrha) [reply src/markspelling.py]: :sparkles: Magic! :sparkles:

Comment (tofu-rocketry) [reply src/markspelling.py]: Should at least add a comment mentioning that they're bash escape codes.

**Review by tofu-rocketry — CHANGES_REQUESTED** [review]

File: src/markspelling.py [hunk 7]
```diff
@@ -18,6 +18,15 @@
 from enchant.checker import SpellChecker
 from enchant.tokenize import EmailFilter, URLFilter
 
+# Format string used to render debug output
+_DEBUG_FORMAT = '%03d %5s %s%s%s'
+
+# ANSI color escape codes for highlighting debug output
```
Comment (tofu-rocketry) [inline src/markspelling.py:24]: "colour"

**Review by tofu-rocketry — CHANGES_REQUESTED** [review]

**Review by tofu-rocketry — APPROVED** [review]

**Pull request merged**